clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -55,19 +72,56 @@ public JmxReporter() {
      * Create a JMX reporter that prefixes all metrics with the given string.
      */
     public JmxReporter(String prefix) {
+        this(prefix, "");
+    }
+
+    public JmxReporter(String prefix, String configPrefix) {
```

Comment:
This doesn't fit with the pattern that we typically use for configuration keys.  Prefixing should be done outside the class in question.  Check out `StreamsConfig#getAdminConfigs` for an example of doing prefixing.

"kafka" is also an odd prefix to use since both the client, broker, etc. all all part of kafka.  If we want to use a prefix just for the broker it should be something like "broker".

Reply:
I'm only following existing conventions here. On the server side we already have the `kafka.` prefix for broker metrics, see [`kafka.metrics.polling.interval.secs`](https://github.com/apache/kafka/blob/1b08fceb7a3c502d2d18dc99796101ed0b4ff8b7/core/src/main/scala/kafka/server/KafkaConfig.scala#L435).

It's not possible to not know the prefix, because in order to implement `reconfigurableConfigs` we have to know the full config name, not just the stripped one, and unfortunately between clients and broker the prefix is different.

Reply:
It's true that there are two kind of weird and old yammer config knobs, `kafka.metrics.reporters` and `kafka.metrics.polling.interval.secs` that are prefixed with "kafka."  But no other broker configurations are.  For example, `metrics.sample.window.ms` isn't prefixed, `metrics.num.samples` isn't prefixed, etc. etc.  And of course, there are hundreds of other broker configurations that are not prefixed.

It doesn't make sense to prefix configurations with "kafka" since logically, every Kafka configuration is for kafka.  Kafka Client configurations are for Kafka, Kafka command line configurations are for Kafka, etc.

Reply:
By the way, `kafka.metrics.reporters` is a horrible config key name because it suggests that it is configuring the "Kafka metrics" system (which, as you know, is separate and different from the Yammer metrics system), but actually no, it configures Yammer.  :disappointed:

Reply:
I'm fine renaming the config, maybe we should consider renaming those "weird, old" metric configs to avoid confusion?

Reply:
Yeah, I think it's worth considering renaming those two configs.  If someone created a KIP for that, I would vote for it.  We would have to support the old names for a while, of course.

Reply:
I've updated the KIP and removed the `kafka.` prefix, which removes the need for any kind of configPrefix.

Reply:
OK, thanks!